mozilla/experimenter
# Simplify integration test workflow fixes #1914

Okay! So I made some changes to the integration test workflow to hopefully make things simpler to use.

Changes:

  - Reuse the standard local dev for running the integration tests instead of a separate copy
  - Change `make kill` to REALLY kill everything it can
  - Setup the vnc server to disable headless mode so you can watch the tests running
  - Had to take out one of the tests because I couldn't think of a good way to get it passing while reusing the local dev instance (@jrbenny35 if you have an idea of how to keep that test around let me know and I'll put it back otherwise I'd rather lose a test to get this workflow working smoothly)

How to test this:
  - Manually kill/remove all your containers with
        `docker ps -aq | xargs docker kill`
        `docker ps -aq | xargs docker rm`
        `docker volume ls -q | xargs docker volume rm`
  - Refresh to create a new db
        `make refresh`
  - Start the regular dev server with either of
        `make up`
        `make up_detached`
  - Check that the local dev is running
        [https://localhost/](https://localhost)
  - Run the headless integration tests in a separate shell
        `make integration_test`
  - Start the vnc server with either of
        `make integration_vnc_up`
        `make integration_vnc_up_detached`
  - Connect to vnc server with password `secret`
        `vnc://localhost:5900`
  - Right click on the linux desktop and go to Applications > Shell > Bash and enter
        `tox -c tests/integration/`
  - This should run all the tests and  you should be able to see the browser running them
  - Close vnc
  - Check your local dev is still running and full of dummy integration test experiments nowo
        [https://localhost/](https://localhost)
  - You should also be able to use @jrbenny35 's instructions to run the tests locally against the same dev instance: https://github.com/mozilla/experimenter/issues/2317


So this way you can keep your local dev running the whole time and then hit it with headless/headful(?) integration tests in any order/combination.  I think this should make using/maintaining/modifying the integration tests a lot simpler hopefully 🤞 

Let me know if it breaks!
